packages/api/src/external/commonwell/link/shared.ts

```diff
@@ -44,6 +47,50 @@ export function patientWithCWData(
   return patientWithCW;
 }
 
+async function shouldDowngradeLink(link: NetworkLink): Promise<boolean> {
+  const identifiers = link.patient?.identifier || [];
+  const checkPromises = identifiers.map(async id => {
+    const idSystem = id.system?.replace(/^urn:oid:/, "");
+    if (idSystem) {
+      return (await getCQDirectoryEntryById(idSystem)) !== undefined;
+    }
+    return false;
+  });
+  const results = await Promise.all(checkPromises);
+  return results.includes(true); // true if any of the checks are true
+}
+
+async function assessIfCWCQDuplicatesAndDowngrade(
```

Comment:
nit: This calls for us to define the all caps for acronym or not. It gets very ugly and hard to read when we need to group them together ([discussion on Notion](https://www.notion.so/Development-Best-Practices-bb3574d3c83d46dc9914a9e5c5542ca7?d=174fbe104e524c1986f2af2d91e1c351&pvs=4#65562fe296794944a864eebd6d7a2e93)).

Can we have only the first letter in caps and the next/remaining ones in lower case? `assessIfCwCqDuplicatesAndDowngrade`

Also, the name doesn't read easily: "assess if CW CQ duplicates and downgrade". If what?

Reply:
`assessIfCwCqDuplicateLinksAndThenDowngradeCwLinks`
Is that better?

Reply:
no longer relevant